src/main/java/eu/dissco/backend/repository/DigitalMediaObjectRepository.java

```diff
@@ -1,103 +1,225 @@
+import lombok.extern.slf4j.Slf4j;
 import org.jooq.DSLContext;
+import org.jooq.Field;
 import org.jooq.Record;
 import org.jooq.Record1;
+import org.jooq.impl.DSL;
 import org.springframework.stereotype.Repository;
 
 @Repository
 @RequiredArgsConstructor
+@Slf4j
 public class DigitalMediaObjectRepository {
 
   private final ObjectMapper mapper;
   private final DSLContext context;
 
   public DigitalMediaObject getLatestDigitalMediaById(String id) {
     return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
-        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID)
-        .from(NEW_DIGITAL_MEDIA_OBJECT)
+        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID).from(NEW_DIGITAL_MEDIA_OBJECT)
         .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
         .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc())
         .fetchOne(this::mapToMultiMediaObject);
   }
 
-  private DigitalMediaObject mapToMultiMediaObject(Record dbRecord) {
-    try {
-      return new DigitalMediaObject(
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.VERSION),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.CREATED),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.TYPE),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.FORMAT),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.SOURCE_SYSTEM_ID),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DATA).data()),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ORIGINAL_DATA).data())
-      );
-    } catch (JsonProcessingException e) {
-      throw new RuntimeException(e);
-    }
+  public JsonApiData getLatestDigitalMediaObjectByIdJsonResponse(String id) {
+
+    Field<Integer> maxSpeciesVersion = DSL.max(NEW_DIGITAL_SPECIMEN.VERSION)
+        .as("maxSpeciesVersion");
+
+    var specimenRecentVersions = context.select(NEW_DIGITAL_SPECIMEN.ID, maxSpeciesVersion)
+        .from(NEW_DIGITAL_SPECIMEN).groupBy(NEW_DIGITAL_SPECIMEN.ID).asTable();
+
+    return context.select(NEW_DIGITAL_SPECIMEN.SPECIMEN_NAME, NEW_DIGITAL_SPECIMEN.VERSION,
+            NEW_DIGITAL_SPECIMEN.ID, NEW_DIGITAL_MEDIA_OBJECT.asterisk()).from(NEW_DIGITAL_SPECIMEN)
+        .join(specimenRecentVersions)
+        .on(NEW_DIGITAL_SPECIMEN.ID.eq(specimenRecentVersions.field(NEW_DIGITAL_SPECIMEN.ID)))
+        .and(NEW_DIGITAL_SPECIMEN.VERSION.eq(specimenRecentVersions.field(maxSpeciesVersion)))
+        .join(NEW_DIGITAL_MEDIA_OBJECT)
+        .on(NEW_DIGITAL_SPECIMEN.ID.eq(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID))
+        .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
+        .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc()).limit(1)
+        .fetchOne(this::mapToJsonApiData);
+  }
+
+  public List<JsonApiData> getAnnotationsOnDigitalMediaObject(String mediaId){
```

Comment:
I would like to get all calls to the Annotations table in the AnnotationRepository. I would say that the DigitalMediaService -> AnnotationService -> AnnotationRepository